resources/core/charts/iam-kubeconfig-service/templates/deployment.yaml

```diff
@@ -32,7 +32,11 @@ spec:
         command:
```

Comment:
Suggestion: If we are doing multiple ifs that all depend on the same condition (isGardener here), it may be better to have 2 depoyment files (deployment and deployment-gardener) and pick which one we use.